### PR TITLE
feat(cli): Export source maps outputs when exporting API route bundles

### PR DESCRIPTION
**Pull request by kitten**
# Why

We previously didn't output source maps when exporting API routes.

# How

- Passthrough existing source map data when exporting server bundles
- Reformat source map paths before doing so (See note)

### CODE REVIEW

File: packages/@expo/cli/src/start/server/metro/MetroBundlerDevServer.ts [hunk 1]
```diff
@@ -131,8 +133,30 @@ export class MetroBundlerDevServer extends BundlerDevServer {
...
+              version: contents.map.version,
+              sources: contents.map.sources.map((source: string) => {
+                return typeof source === 'string' && source.startsWith(this.projectRoot)
+                  ? path.relative(this.projectRoot, source)
+                  : source;
+              }),
+              sourcesContent: new Array(contents.map.sources.length).fill(null),
+              names: contents.map.names,
+              mappings: contents.map.mappings,
+            }),
```
Comment (kitten) [inline packages/@expo/cli/src/start/server/metro/MetroBundlerDevServer.ts:154]: This is something I'd ideally would want to get rid of 🤔 

The problem here is that we need `sources` to be absolute for `source-map-support` (at least, as far as I know), but the output needs to be relative to the project root again.

It's potentially possible to see whether we can change the output to contain the absolute path to `sourceRoot`, but it's unclear whether `sourceRoot` will be ignored on deployed platforms.

File: packages/@expo/cli/src/start/server/metro/MetroBundlerDevServer.ts [hunk 2]
```diff
@@ -131,8 +133,30 @@ export class MetroBundlerDevServer extends BundlerDevServer {
         path.relative(appDir, filepath.replace(/\.[tj]sx?$/, '.js'))
       );
       if (contents) {
+        let src = contents.src;
```
Comment (EvanBacon) [inline packages/@expo/cli/src/start/server/metro/MetroBundlerDevServer.ts:136]: Add a TODO about merging this with the other source map treatment in the future https://github.com/expo/expo/blob/0dffdb15d9f3b3053458023b54153cb7a5f92b73/packages/%40expo/metro-config/src/serializer/serializeChunks.ts#L422-L439

**Review by EvanBacon — APPROVED** [review]
Add TODO and also update the CLI E2E test to account for the source maps existing and ensuring they have relative paths.

**Review by EvanBacon — APPROVED** [review]

**Review by Missjones2829 — COMMENTED** [review]
packages/@expo/cli/e2e/__tests__/export/server.test.ts

**Pull request merged**